File: RegressionTests/Compiler/Sources/TestForEach.cs
using Java.Util;
using Junit.Framework;

namespace Dot42.Tests.Compiler.Sources
{
    public class TestForEach : TestCase
    {
        public void testIntArray()
        {
            var arr = new[] { 1, 2, 3, 4, 5, 6 };
            var index = 0;
            foreach (var value in arr)
            {
                AssertEquals(index + 1, value);
                index++;
            }
            AssertEquals(index, 6);
        }

        public void testStringArray()
        {
            var arr = new[] { "a", "bb", "ccc", "dddd" };
            var index = 0;
            foreach (var value in arr)
            {
                AssertEquals(index + 1, value.Length);
                index++;
            }
            AssertEquals(index, 4);
        }

        public void testStringList()
        {
            var arr = new ArrayList<string>();
            arr.Add("a");
            arr.Add("bb");
            arr.Add("ccc");
            arr.Add("dddd");
            var index = 0;
            foreach (var value in arr)
            {
                AssertEquals(index + 1, value.Length);
                index++;
            }
            AssertEquals(index, 4);
        }
    }
}

File: RegressionTests/JavaCompiler/JavaSources/TestForEach.java
package dot42.java.test;

import java.util.*;
import junit.framework.*;

public class TestForEach extends TestCase
{
	public void testIntArray()
	{
		int[] arr = new int[] { 1, 2, 3, 4, 5, 6 };
		int index = 0;
		for (int value : arr)
		{
			assertEquals(index + 1, value);
			index++;
		}
		assertEquals(index, 6);
	}

	public void testStringArray()
	{
		String[] arr = new String[] { "a", "bb", "ccc", "dddd" };
		int index = 0;
		for (String value : arr)
		{
			assertEquals(index + 1, value.length());
			index++;
		}
		assertEquals(index, 4);
	}

	public void testStringList()
	{
		ArrayList<String> arr = new ArrayList<String>();
		arr.add("a");
		arr.add("bb");
		arr.add("ccc");
		arr.add("dddd");
		int index = 0;
		for (String value : arr)
		{
			assertEquals(index + 1, value.length());
			index++;
		}
		assertEquals(index, 4);
	}
}
